browser/skus/skus_service_factory.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/browser/skus/skus_service_factory.h"
+
+#include "brave/browser/profiles/profile_util.h"
+#include "brave/components/skus/browser/skus_utils.h"
+#include "chrome/browser/profiles/profile.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/storage_partition.h"
+
+namespace skus {
+
+// static
+SkusServiceFactory* SkusServiceFactory::GetInstance() {
+  return base::Singleton<SkusServiceFactory>::get();
+}
+
+// static
+mojo::PendingRemote<mojom::SkusService> SkusServiceFactory::GetForContext(
+    content::BrowserContext* context) {
+  return static_cast<skus::SkusService*>(
+             GetInstance()->GetServiceForBrowserContext(context, true))
+      ->MakeRemote();
+}
+
+// static
+void SkusServiceFactory::BindForContext(
+    content::BrowserContext* context,
+    mojo::PendingReceiver<skus::mojom::SkusService> receiver) {
+  auto* service = static_cast<skus::SkusService*>(
+      GetInstance()->GetServiceForBrowserContext(context, true));
+  if (service) {
+    service->Bind(std::move(receiver));
+  }
+}
+
+SkusServiceFactory::SkusServiceFactory()
+    : BrowserContextKeyedServiceFactory(
+          "SkusService",
+          BrowserContextDependencyManager::GetInstance()) {}
+
+SkusServiceFactory::~SkusServiceFactory() = default;
+
+KeyedService* SkusServiceFactory::BuildServiceInstanceFor(
+    content::BrowserContext* context) const {
+  // Skus functionality not supported in private / Tor / guest windows
+  if (!brave::IsRegularProfile(context)) {
```

Comment:
you also need to check the feature flag here and return nullptr

Reply:
Fixed in https://github.com/brave/brave-core/pull/10847/commits/bc08735f11dc33351fe3845cef9a29959b4e6eb4